src/api/code-challenge/content-types/code-challenge/test-evaluator.ts

```diff
@@ -0,0 +1,279 @@
+import {
+  evaluateWithContext,
+  getCode,
+  getEvaluationContext,
+  overrideConsoleLog,
+  restoreConsoleLog,
+} from "../code-challenge/utils";
+
+type runTestEvaluatorProps = {
+  metaCaseCode: string;
+  internalTestCode: string;
+  metaLabel: string;
+  challengeLabel: string;
+  metaTestId: number;
+  evalResultShouldBe: boolean;
+  internalTest?: object;
+  removeComments?: boolean;
+};
+
+export const runTestEvaluator = async ({
+  metaCaseCode,
+  internalTestCode,
+  metaLabel,
+  challengeLabel,
+  metaTestId,
+  evalResultShouldBe,
+  internalTest,
+  removeComments,
+}: runTestEvaluatorProps) => {
+  let userPassed = null;
+  let userShouldPass = evalResultShouldBe;
+  let evaluationError;
+  let descriptionMessage;
+  // let tryBlockResult;
+  const formattedCode = getCode(metaCaseCode, removeComments);
+  // console.log(
+  //   `-----
+  //   \n<debug internalTest-evaluator.ts>
+  //   \n PARAM TYPE: <${typeof internalTest}>,
+  //   \n <TEST PARAM>${JSON.stringify(internalTest)}
+  //   \n-------`
+  // );
+  const logs = [] as Array<unknown>;
+  // Checking Params in Finally Block too
+
+  overrideConsoleLog((args) => {
+    // Disabled for easier console reading
+    logs.push(args);
+    // @ts-expect-error will fix later
+    console.standardLog("args", ...args);
+  });
+  console.log("\n<challengeLabel>", challengeLabel);
+  console.log("\n\n<<METATEST CODE>>", metaCaseCode);
+  console.log("<formattedCode>", formattedCode);
+  console.log("<formattedCode TYPE>", typeof formattedCode);
+  console.log("<internalTestCode>", internalTestCode, "\n\n");
+  let result;
+
+  // Idea: Figure out programmatically whether metaCode will throw an error.
+  // Tomorrow: make a bool for metaCaseCode errors, if true, allow undefined === userPassed
+
+  // Apr 12
+  // Logic: if visibleMetaError !== null, set metaError === true, if metaError === true, undefined <FINALLY result> is a userError
+  // logic not yet tested on multiple code-challenges
+  let visibleMetaError = null;
+  let existsMetaError = false;
+  try {
+    const metaCheck = eval(formattedCode);
+    console.log("<metaCheck>", metaCheck);
+  } catch (err) {
+    // console.log("<metaCaseCode ERROR>", err);
+    visibleMetaError = err;
+    existsMetaError = true;
+    // if metaError exists, result should be undefined
+    // if no metaError, result should be boolean
+  } finally {
+    console.log("<metaCaseCode ERROR>", visibleMetaError);
+    console.log("<existsMetaError>", existsMetaError);
+  }
```

Comment:
This is written in a way that would confuse someone if they were unfamiliar with this code. I am still not sure we are going to keep all this extra logic. It depends on how clear the code is in it's final, cleaned up form. However, the next step for cleanup is can you extract this into some sort of function that returns a result? That way the logic is encapsulated and labelled

Reply:
Summary: We spoke on the phone and decided to drop the "failing tests"-logic (code example above) for this PR.